### PR TITLE
Async TP to another entity

### PR DESCRIPTION
**Pull request by JT122406**
Can easily be done by devs but why not have it directly in the lib

### CODE REVIEW

File: src/main/java/io/papermc/lib/features/asyncteleport/AsyncTeleportSync.java [hunk 1]
```diff
@@ -13,4 +13,9 @@ public CompletableFuture<Boolean> teleportAsync(Entity entity, Location location
         return CompletableFuture.completedFuture(entity.teleport(location, cause));
     }
 
+    @Override
+    public CompletableFuture<Boolean> teleportAsync(Entity entity, Entity entity2, TeleportCause cause) {
+        return CompletableFuture.completedFuture(entity.teleport(entity2.getLocation(), cause));
```
Comment (yannicklamprecht) [inline src/main/java/io/papermc/lib/features/asyncteleport/AsyncTeleportSync.java:18]:
```suggestion
        return this. teleportAsync(entity, entity2.getLocation(), cause);
```

File: src/main/java/io/papermc/lib/features/asyncteleport/AsyncTeleportPaper.java [hunk 2]
```diff
@@ -16,4 +16,12 @@ public CompletableFuture<Boolean> teleportAsync(Entity entity, Location location
         return PaperLib.getChunkAtAsyncUrgently(location.getWorld(), x, z, true).thenApply(chunk -> entity.teleport(location, cause));
     }
 
+    @Override
+    public CompletableFuture<Boolean> teleportAsync(Entity entity, Entity entity2, TeleportCause cause) {
+        Location location = entity2.getLocation();
+        int x = location.getBlockX() >> 4;
+        int z = location.getBlockZ() >> 4;
+        return PaperLib.getChunkAtAsyncUrgently(location.getWorld(), x, z, true).thenApply(chunk -> entity.teleport(location, cause));
```
[the remark this answers is not in the document]
Comment (yannicklamprecht) [reply src/main/java/io/papermc/lib/features/asyncteleport/AsyncTeleportPaper.java:24]:
```suggestion
        return this.teleportAsync(entity, entity2.getLocation(), cause);
```

I See no reason in duplication code if the only diff is the location.

**Review by yannicklamprecht — CHANGES_REQUESTED** [review]
Basically a nonsense PR if you can just call entity2.getLocation() and pass the result to the previous method.

**Comment by JRoy** [comment]
This is just API bloat. The difference would be calling teleportAsync(entity) vs teleportAsync (entity.getLocation()) which is hardly an inconvenience.

**Pull request closed without merging**